client/src/components/MultiJurisdictionAudit/Progress/ProgressMap.tsx

```diff
@@ -0,0 +1,367 @@
+  }
+  &.progress {
+    background-color: ${Colors.ORANGE4};
+  }
+  &.gray {
+    background-color: ${Colors.GRAY4};
+  }
+`
+
+const MapSpinner = styled(Spinner)`
+  position: absolute;
+  top: 50%;
+  right: 0;
+  left: 0;
+  margin: 0 auto;
+`
+
+const Map = ({
+  stateName,
+  jurisdictions,
+  isRoundStarted,
+  auditType,
+}: IProps) => {
+  const width = 960
+  const height = 500
+  const d3Container = useRef(null)
+  const tooltipContainer = useRef(null)
+
+  const [usCounties, setUSCounties] = useState<Feature[] | undefined>(undefined)
+  const [usStates, setUSStates] = useState<Feature[] | undefined>(undefined)
+  const [usState, setUSState] = useState<Feature | undefined>(undefined)
+  const [jsonData, setJsonData] = useState<Topology | undefined>(undefined)
+  const [isMapLoaded, setIsMapLoaded] = useState<boolean>(false)
+
+  const getStateName = (abbr: string) => {
+    const filteredState = labelValueStates.find(
+      ({ value, label }) => value === abbr && label
+    )
+    return filteredState && filteredState.label
+  }
+
+  const getJurisdictionStatus = useCallback(
+    (countyName: string) => {
+      const filteredJurisdiction = jurisdictions.find(
+        jurisdiction => jurisdiction.name === countyName
+      )
+
+      if (filteredJurisdiction) {
+        const {
+          currentRoundStatus,
+          ballotManifest,
+          batchTallies,
+          cvrs,
+        } = filteredJurisdiction
+
+        if (!currentRoundStatus) {
+          const files: IFileInfo['processing'][] = [ballotManifest.processing]
+          if (batchTallies) files.push(batchTallies.processing)
+          if (cvrs) files.push(cvrs.processing)
+
+          const numComplete = files.filter(
+            f => f && f.status === FileProcessingStatus.PROCESSED
+          ).length
+          const anyFailed = files.some(
+            f => f && f.status === FileProcessingStatus.ERRORED
+          )
+
+          // Special case when we just have a ballotManifest
+          if (files.length === 1) {
+            if (anyFailed) {
+              return 'danger'
+            }
+            if (numComplete === 1) {
+              return 'success'
+            }
+            return 'gray'
+          }
+
+          // When we have multiple files
+          if (anyFailed) {
+            return 'danger'
+          }
+          if (numComplete === files.length) {
+            return 'gray'
+          }
+        } else {
+          if (currentRoundStatus.status === JurisdictionRoundStatus.COMPLETE) {
+            return 'success'
+          }
+          if (
+            currentRoundStatus.status === JurisdictionRoundStatus.IN_PROGRESS
+          ) {
+            return 'progress'
+          }
+          if (
+            currentRoundStatus.status === JurisdictionRoundStatus.NOT_STARTED ||
+            currentRoundStatus.status === null
+          ) {
+            return 'gray'
+          }
+        }
+      }
+      return 'default'
```

Comment:
since we want to ensure the progress table and the map always show the same status, it think it would be good to factor out the logic that determines the status. i'm imagining creating a function that takes a jurisdiction and returns the status that we want to show in the table/map